internal/controller/configmap.go

```diff
@@ -0,0 +1,419 @@
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package controller
+
+import (
+	"context"
+	"fmt"
+
+	corev1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	argoprojiov1alpha1 "github.com/argoproj-labs/argocd-rbac-operator/api/v1alpha1"
+	"github.com/argoproj-labs/argocd-rbac-operator/internal/controller/common"
+)
+
+// createRBACConfigMap will create the Argo CD RBAC ConfigMap resource.
+func (r *ArgoCDRoleReconciler) createRBACConfigMap(cm *corev1.ConfigMap, role *argoprojiov1alpha1.ArgoCDRole) error {
+	data := make(map[string]string)
+	overlayKey := fmt.Sprintf("policy.%s.csv", role.ObjectMeta.Name)
+	roleName := fmt.Sprintf("role:%s", role.ObjectMeta.Name)
+
+	data[common.ArgoCDKeyRBACPolicyDefault] = getRBACPolicyDefault()
+	data[common.ArgoCDKeyRBACScopes] = getRBACScopes()
+	data[common.ArgoCDKeyRBACPolicyCSV] = getDefaultRBACPolicy()
+	data[overlayKey] = buildPolicyStringRules(role, roleName)
+	cm.Data = data
+
+	return r.Client.Create(context.TODO(), cm)
+}
+
+// getRBACPolicyDefault will return the Argo CD RBAC default policy.
+func getRBACPolicyDefault() string {
+	return common.ArgoCDDefaultRBACDefaultPolicy
+}
+
+// getRBACScopes will return the Argo CD RBAC scopes.
+func getRBACScopes() string {
+	return common.ArgoCDDefaultRBACScopes
+}
+
+// getRBACDefaultPolicyCSV will return the Argo CD RBAC default policy CSV.
+func getDefaultRBACPolicy() string {
+	return common.ArgoCDDefaultRBACPolicy
+}
+
+func getRBACPolicyCSV(role *argoprojiov1alpha1.ArgoCDRole, rb *argoprojiov1alpha1.ArgoCDRoleBinding) string {
+	policy := ""
+	roleName := fmt.Sprintf("role:%s", role.ObjectMeta.Name)
+
+	policy += buildPolicyStringRules(role, roleName)
+	policy += buildPolicyStringSubjects(rb, role)
+
+	return policy
+}
+
+// buildPolicyStringRules will build the policy string for Rules field of the given role.
+func buildPolicyStringRules(role *argoprojiov1alpha1.ArgoCDRole, roleName string) string {
+	policy := ""
+	for _, rule := range role.Spec.Rules {
+		resource := rule.Resource
+		for _, verb := range rule.Verbs {
+			for _, object := range rule.Objects {
+				policy += fmt.Sprintf("p, %s, %s, %s, %s, allow\n", roleName, resource, verb, object)
+			}
+		}
+	}
+	return policy
+}
+
+// buildPolicyStringSubjects will build the policy string for Subjects field of the given role.
+func buildPolicyStringSubjects(rb *argoprojiov1alpha1.ArgoCDRoleBinding, role *argoprojiov1alpha1.ArgoCDRole) string {
```

Comment:
I think the unit test coverage should be near 100% for the initial commit. A lot of e2e/ unit tests are missing make it hard to asses the validity of the code.

Reply:
I'll work on some unittests

Reply:
I have opened a new issue #3 to implement the unit tests.